### ISSUE TITLE
gap CSS property incorrectly adds gaps for absolutely positioned children

### ISSUE THREAD

**Issue by lexi-stein** [facebook/yoga#1652]
### Description

In HTML/CSS, `gap` applies gutters between child elements but ignores absolutely positioned children. On iOS React Native, `gap` also adds gutters for each absolutely positioned child, unexpectedly increasing the size of the parent container.

### React Native Version

0.71.6

### Output of `npx react-native info`

System:
    OS: macOS 13.4.1
    CPU: (10) x64 Apple M1 Pro
    Memory: 22.36 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.17.0/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.17.0/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_131 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Create a container View with a `gap` set (larger ones will make it easier to see the problem).
Give it two children, and make one of them absolutely positioned.
Notice the gap added incorrectly.

### Snack, code example, screenshot, or link to a repository

```
<View style={{ gap: 12, backgroundColor: 'blue' }}>
  <View style={{ backgroundColor: 'red', width: 20, height: 20 }} />
  <View style={{ position: 'absolute' }} />
</View>
```
<img width="21" alt="image" src="https://github.com/facebook/react-native/assets/33403793/156c6dbc-f82f-4ddc-8fd6-da92875ed25c">

As compared to
```
<div style={{ gap: 12, backgroundColor: 'blue' }}>
  <div style={{ backgroundColor: 'red', width: 20, height: 20 }} />
  <div style={{ position: 'absolute' }} />
</div>
```
<img width="32" alt="image" src="https://github.com/facebook/react-native/assets/33403793/bc7d4ecd-0e1b-41b2-987b-681c72e735d7">

**Comment by NickGerleman** [comment]
> I've noticed in the snack and in my real use-case that changing the order of views makes a difference. No idea why.
> 
> Also seems like this changes based on version. Setting Expo version `50.0` in the snack seems to work as expected.
> 
> https://snack.expo.dev/BsdbOBR_qRBpVz1crwvJp

https://github.com/facebook/yoga/commit/c60050d0cb248a7f495ef79538145438ab0955fd fixed a bug around here, included in Yoga 3.0/RN 0.74/Expo 50.

Would be helpful to know if folks still repro on this version.

**Comment by NickGerleman** [comment]
The example in the original issue seems to no longer repro in Yoga Playground. https://www.yogalayout.dev/playground?code=DwGQhgng9grgLgAgMZQHYDMCWBzAvAb3xgGcBTAdVICMARU9MGAGzmIC4EGmyBfHgPgBQCBMAByUACakExOBCakC+BNjAAHDgEYATABoEVMEgDW2AE6xUkgMJQmUcxwDkVJjFLOEfISJHipGTkFJUJDYzNLGGs7BycEZ3NSSWcDAHdMSTgACw4dAAYDbNIcbLg8-O8eBAB6Xz8A6Vl5RWUEdShiTDhMNBcwKmJ7eE8q2vrgGolpIUnwaHh+IA

Caveated, RN opts out of some of the newer code for absolute positioning for compatibility reasons, that the playground currently does not.
